Rakefile

```diff
@@ -0,0 +1,67 @@
+# Rakefile
+# Copyright (C) 2018 Mike JS Choi
+# Copyright (C) 2018 VLC authors and VideoLAN
+# $Id$
+#
+# Authors: Mike JS. Choi <mkchoi212 # icloud.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+#
+#
+#
+# ------------------------------------------------------------- Constants ------
+
+SDK_SIM = "iphonesimulator11.3"
+SDK_SIM_DEST = "'platform=iOS Simulator,name=iPhone 7,OS=11.3'"
+
+SCHEME_IOS = "MobileVLCKit"
+PROJECT_IOS = "MobileVLCKit.xcodeproj"
+
+VLC_FLAGS_SIM = "-da x86_64"
+
+# ----------------------------------------------------------------- Tasks ------
+
+desc "Build VLCKit (iOS Simulator)"
+task "build:vlckit:ios" do
+  puts "Building VLCKit (iOS Simulator)"
+
+  plugin_file = "Resources/MobileVLCKit/vlc-plugins-iPhone.h"
+  required_dirs = ["./libvlc/vlc/install-iPhone", "./libvlc/vlc/install-iPhoneSimulator", "./libvlc/vlc/build-iPhoneSimulator"]
+
+  if File.exist?(plugin_file) and dirs_exist?(required_dirs)
+    puts "Found pre-existing build directory. Skipping build"
```

Comment:
I actually think you don't need to check for this since the compileAndBuild will be already fast if those directories exist

Reply:
Hmmm on my machine, it takes about 15 minutes to do a build even when those directories exist 🤔

Reply:
Also I'm not 100% how it works yet but I think every time a CI build is triggered, it clones a fresh version of the repo, therefore removing the previous build directories

Reply:
Yes, that's actually true you would need to build a cache yourself. We would need to make sure that we don't use the cache when we add for example a new patch or change something in the compileAndBuildVLCKit.sh

Reply:
If `compileAndBuildVLCKit.sh` changes, CircleCI will do a rebuild. It's done in `.circle.ci/config.yml` with this 
```
- save_cache:
    key: libvlc-{{ checksum "compileAndBuildVLCKit.sh" }}
```
I'm not sure about adding new patches though cause I haven't really figured out how patches work yet :p